pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -151,14 +126,31 @@ patchShebangs() {
 
                 # Preserve times, see: https://github.com/NixOS/nixpkgs/pull/33281
                 timestamp=$(stat --printf "%y" "$f")
-                sed -e "1 s|.*|#\!$escapedInterpreterLine|" "$f" | _sponge "$f"
+
+                # Manually create temporary file instead of using sed -i
+                # (sed -i on $out/x creates tmpfile /nix/store/x which fails on macos + sandbox)
+                tmpFile=$(mktemp -t patchShebangs.XXXXXXXXXX)
+                sed -e "1 s|.*|#\!$escapedInterpreterLine|" "$f" > "$tmpFile"
+
+                # Make original file writable if it is read-only
+                local restoreReadOnly
+                if [[ ! -w "$f" ]]; then
+                    chmod +w "$f"
```

Comment:
Should this maybe also track executable-ness?

Reply:
Since we are writing back to the original file, there is no need to track it. Only the writable flag needs to be restored, because the file needs to be made writable in order to update it